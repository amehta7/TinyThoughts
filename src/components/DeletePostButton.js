import React from 'react'
import PropTypes from 'prop-types'

const DeletePostButton = () => {
  return <div className='delete-post-btn'></div>
}

export default DeletePostButton
