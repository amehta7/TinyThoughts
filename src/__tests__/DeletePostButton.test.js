import React from "react";
import { mount } from "enzyme";
import DeletePostButton from "../components/DeletePostButton";

describe("DeletePostButton component", () => {
  it("DeletePostButton component renders & works correctly", () => {
    const onConfirmHandler = jest.fn();
    const wrapper = mount(<DeletePostButton onConfirm={onConfirmHandler} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find(".delete-btn").simulate("click");
    expect(wrapper).toMatchSnapshot();
    wrapper.find(".delete-btn-yes").simulate("click");
    expect(onConfirmHandler).toHaveBeenCalled();
  });
});
