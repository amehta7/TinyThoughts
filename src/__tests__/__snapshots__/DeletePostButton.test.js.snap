// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePostButton component DeletePostButton component renders & works correctly 1`] = `
<DeletePostButton
  onConfirm={[MockFunction]}
>
  <div
    className="delete-post-btn"
  >
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      Delete?
    </button>
  </div>
</DeletePostButton>
`;

exports[`DeletePostButton component DeletePostButton component renders & works correctly 2`] = `
<DeletePostButton
  onConfirm={[MockFunction]}
>
  <div
    className="delete-post-btn"
  >
    <button
      className="delete-btn-yes"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="delete-btn-no"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</DeletePostButton>
`;
